### ISSUE TITLE
Move code from index.js files into files named after components

### ISSUE THREAD

**Issue by nkohari** [auth0/cosmos#372]
Our current file naming convention is (for example):

```
text-input/
  index.js
  text-input.md
```

Since the actual implementation of the components is always stored in a file called `index.js`, it can be difficult to navigate the codebase.

Instead of doing this, we should move to the following convention:

```
text-input/
  text-input.js
  text-input.md
  index.js
```

Where `index.js` is just a stub file which imports and exports:

```js
export { TextInput } from './text-input'
```

This has the following advantages:
1. It becomes easier to jump to a component file in the editor.
2. The name of the documentation matches the name of the code file.
3. If the component has sub-components, it arguably is more logical to split them off into separate files using this system.

**Comment by landitus** [comment]
I agree, seems like the only disadvantege is one extra file but there's not much to maintain after that.

I've suffered this: I use a plugin in Atom to show the folder in the tab so I know which component I'm working on!

<img width="414" alt="screen shot 2018-03-07 at 16 10 43" src="https://user-images.githubusercontent.com/239215/37112824-5e334906-2222-11e8-9d76-7d6baad1a477.png">

**Comment by nkohari** [comment]
Closed by #547.
